docs/source/_ext/atc.py

```diff
@@ -46,6 +46,29 @@ def run(self):
 			impl_node.append(nodes.paragraph('', '', *(n + m)))
 		return [impl_node]
 
+# -- Go Version role --------------------------------------------------
+# Returns the value of the Go version stored in GO_VERSION to minor version
+# precision.
+import os
+import re
+
+from docutils.nodes import strong
+from docutils.parsers.rst.states import Inliner
+from typing import Tuple, List
+
+def atc_go_version(unused_typ: str,
+               unused_rawtext: str,
+               unused_text: str,
+               unused_lineno: int,
+               unused_inliner: Inliner,
+               options=None,
+               content=None) -> Tuple[List[strong], list]:
+	go_version_file = os.path.join(os.path.dirname(__file__), '../../../GO_VERSION')
+	go_version = open(file=go_version_file).read()
```

Comment:
Looks like this file is never closed. I'd recommend doing a context-managed read like:
```python
with open(go_version_file) as f:
    go_version = f.read()
    # ...
```
So that the file is always closed and fds don't leak

Reply:
Reading with *with* in 6483fa1d46